File: interviewbit/math/fizzbuzz.java
/*
    interviewbit
    track: "programming",
    section: "math",
    problem title: "FizzBuzz"
*/
public class Solution {
    public ArrayList<String> fizzBuzz(int A) {
        ArrayList<String> fbList = new ArrayList<String>();
        for(int i=1;i<=A;i++){
            StringBuffer entry = new StringBuffer("");
            if(i%3==0){
             entry.append("Fizz");   
            }
            if(i%5==0){
                entry.append("Buzz");
            }
            if(!(i%3==0 || i%5==0)){
                entry.append(Integer.toString(i));
            }
            
            fbList.add(entry.toString());
        }
        
        return fbList;
        
    }
}

File: interviewbit/math/fizzbuzz.cpp
/*
	interviewbit
	track: "programming"
	section: "math"
	problem title: "FizzBuzz"
*/
#include<sstream>
#include<string>

string fizzBuzzOutput(int &num){
    ostringstream strOutput;
    if(num%3==0){
        strOutput<<"Fizz";
    }
    if(num%5==0){
        strOutput<<"Buzz";
    }
    if(num%3!=0 && num%5!=0){
        strOutput<<num;
    }
    return strOutput.str();
}

vector<string> Solution::fizzBuzz(int A) {
    vector<string> outputVector;
    for(int i=1;i<=A;i++){
        outputVector.push_back(fizzBuzzOutput(i));
    }
    return outputVector;
}
